app/scripts/lib/transaction/eip5792.ts

```diff
@@ -109,16 +125,39 @@ export async function processSendCalls(
   const securityAlertId = generateSecurityAlertId();
   const validateSecurity = validateSecurityHook.bind(null, securityAlertId);
 
-  const { batchId: id } = await addTransactionBatch({
-    from,
-    networkClientId,
-    origin,
-    securityAlertId,
-    transactions,
-    validateSecurity,
-  });
+  let batchId: Hex;
+  if (Object.keys(transactions).length === 1) {
+    const trxnParams = {
+      from,
+      ...transactions[0].params,
+      type: TransactionEnvelopeType.feeMarket,
+    };
+    const securityRequest: ValidateSecurityRequest = {
+      method: 'eth_sendTransaction',
+      params: [trxnParams],
+      origin,
+    };
+    validateSecurity(securityRequest, dappChainId);
+    batchId = generateBatchId();
+    await addTransaction(trxnParams, {
+      networkClientId,
+      origin,
+      securityAlertResponse: { securityAlertId } as SecurityAlertResponse,
+      batchId,
+    });
+  } else {
```

Comment:
Minor, rather than this inline logic, could we benefit from a separate function such as `processSingleTransaction` that returns the batch ID?